Dockerfile

```diff
@@ -1,65 +1,102 @@
 MAINTAINER PhenoMeNal-H2020 Project <phenomenal-h2020-users@googlegroups.com>
 
-LABEL Description="Galaxy 17.05-phenomenal for running inside Kubernetes."
+LABEL Description="Galaxy 17.09-phenomenal for running inside Kubernetes."
 LABEL software="Galaxy"
-LABEL software.version="17.05-pheno-lr"
-LABEL version="1.4"
+LABEL software.version="17.09-pheno-lr"
+LABEL version="1.5"
 
 RUN apt-get -qq update && apt-get install --no-install-recommends -y apt-transport-https software-properties-common wget && \
     apt-get update -qq && \
-    apt-get install --no-install-recommends -y mercurial python-psycopg2 sudo python-virtualenv \
-    libyaml-dev libffi-dev libssl-dev \
-    curl git python-pip python-gnuplot python-psutil && \
+    apt-get install --no-install-recommends -y \
+              curl \
+              git \
+              libffi-dev \
+              libssl-dev \
+              libyaml-dev \
+              mercurial \
+              python-gnuplot \
+              python-pip \
+              python-psutil \
+              python-psycopg2 \
+              python-virtualenv \
+              gcc-4.6 \
+              python-dev \
+              sudo \
+    && \
     pip install --upgrade pip && \
     apt-get purge -y software-properties-common && \
     apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-RUN git clone --depth 1 --single-branch --branch rel_17.05_plus_k8s_fs_support_resource_limit_reqs https://github.com/phnmnl/galaxy.git
-WORKDIR galaxy
+
+# Clone galaxy into /galaxy directory
+RUN git clone --depth 1 --single-branch --branch release_17.09_plus_isa_k8s_resource_limts https://github.com/phnmnl/galaxy.git
+WORKDIR /galaxy
+
 RUN echo "pykube==0.15.0" >> requirements.txt
-COPY config/galaxy.ini config/galaxy.ini
-COPY config/job_conf.xml config/job_conf.xml
-COPY config/tool_conf.xml config/tool_conf.xml
-COPY config/sanitize_whitelist.txt config/sanitize_whitelist.txt
-COPY config/job_resource_params_conf.xml config/job_resource_params_conf.xml
-COPY config/phenomenal_tools2container.yaml config/phenomenal_tools2container.yaml
 
-# Galaxy tours which guide users through the subsequent steps in an analysis
-COPY config/plugins/tours/metfrag+xcms-tour.yaml config/plugins/tours/metfrag+xcms-tour.yaml
+COPY config/datatypes_conf.xml \
+     config/galaxy.ini \
+     config/job_conf.xml \
+     config/job_resource_params_conf.xml \
+     config/phenomenal_tools2container.yaml\
+     config/sanitize_whitelist.txt \
+     config/tool_conf.xml \
+  config/
 
 COPY rules/k8s_destinations.py lib/galaxy/jobs/rules/k8s_destination.py
 COPY tools/phenomenal tools/phenomenal
 
 RUN virtualenv .config_script_venv
 RUN /bin/bash -c "source .config_script_venv/bin/activate && \
-                  pip install bioblend==0.9.0 && \
+                  pip install bioblend>=0.9.0 && \
                   deactivate"
 
+RUN ln -s /usr/bin/x86_64-linux-gnu-gcc-4.6 /usr/bin/x86_64-linux-gnu-gcc
```

Comment:
The whole process of install gcc, compiling/installing isatools, and uninstalling gcc needs and dev-libs needs to go in a single RUN statement. Otherwise, we are adding around 500 MBs for free to our container (which weights around 650 MB, so quite a lot!).

Reply:
What happened with ISATools lite that at some point was mentioned? Because we are using the entire ISATools here, right?

Reply:
No, it's isatools-lite, as you can see it in the `git clone` line:
```dockerfile
RUN git clone --depth 1 --single-branch --branch py2_isatools-lite https://github.com/ISA-tools/isa-api.git /isatools
```
I've tried to install gcc after the main `apt-get` call, must it made bioblend installation to fail for some unknown reason. I'll try again, but this time by installing gcc in the same RUN as isatools installation.

Reply:
I've tried that:
```dockerfile
RUN apt-get -qq update && apt-get install --no-install-recommends -y apt-transport-https software-properties-common wget && \
    apt-get update -qq && \
    apt-get -y --no-install-recommends install gcc-4.6 python-dev && \
	ln -s /usr/bin/x86_64-linux-gnu-gcc-4.6 /usr/bin/x86_64-linux-gnu-gcc && \
	/bin/bash -c "source .venv/bin/activate && \
                  pip install 'pip>=8.1' && \
                  pip install -r requirements.txt \
                      --index-url https://wheels.galaxyproject.org/simple \
                      --extra-index-url https://pypi.python.org/simple && \
                  pip install -e /isatools && \
                  deactivate" && \
	apt-get -y remove python-dev gcc-4.6 && \
    apt-get purge -y software-properties-common && \
    apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
But I get this error:
```
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/universe/binary-amd64/Packages  Bad header line [IP: 91.189.88.162 80]

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/multiverse/binary-amd64/Packages  Bad header line [IP: 91.189.88.162 80]

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Reply:
Trying a variant now, give me a sec...

Reply:
I have commited a variant for this eec218fd6c3a34f6c7be795e71e343fba0dbfaed which builds correctly on my local machine.

Reply:
Please pull and try it @pkrog